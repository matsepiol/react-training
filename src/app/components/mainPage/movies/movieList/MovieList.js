import React from "react";

require('./style.scss');

export default class MovieList extends React.Component {


  handleChange() {
    console.log(this);
    this.done = !this.done;
  }

  render() {
    let movie;

    this.props = {
      movie: [{
        name: "Mock movie",
        genre: "unknown",
        year: 2002
      }]
    }

    if (this.props && this.props.movie) {
      movie = this.props.movie.map( (movie, i) => {
        return (
          <div className='movie' key={i}> 
            <p className='movie-name'>Name: {movie.title}</p> 
            <p className='movie-genre'>Genre: {movie.genre}</p> 
            <p className='movie-year'>Year: {movie.year}</p> 
          </div>
        );
      })
    }
    return(
      <div className='movie-wrapper col-lg-12'>{movie}</div>
    )
  }
}


