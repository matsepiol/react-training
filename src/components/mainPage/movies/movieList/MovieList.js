import React from "react";

require('./style.scss');

export default class MovieList extends React.Component {

  componentDidMount() {
    fetch('http://react-cdp-api.herokuapp.com/movies')
    .then(res => res.json())
    .then(
      (result) => {
        console.log(result);
        this.setState({
          isLoaded: true,
          movies: result.data
        });
      },
    )
  }

  render() {
    let movie;

    if (this.state && this.state.movies) {
      console.log(this.state);
      movie = this.state.movies.map( (movie, i) => {
        return (
          <div className='movie' key={i}> 
            <p className='movie-name'>Name: {movie.title}</p> 
            <p className='movie-genre'>Genre: {movie.genre}</p> 
          </div>
        );
      })
    }
    return(
      <div className="movies-container">{movie}</div>
    )
  }
}


