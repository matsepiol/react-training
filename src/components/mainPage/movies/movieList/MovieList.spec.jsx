import * as React from "react";
import { shallow } from "enzyme";
import {create as render } from 'react-test-renderer';

import MovieList from './MovieList';


it("renders correctly", () => {
    const result = render(<MovieList />);
    const snap = result.toJSON();

    expect(snap).toMatchSnapshot();
});
