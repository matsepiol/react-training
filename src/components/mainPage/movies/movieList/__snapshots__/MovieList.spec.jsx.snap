// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders genre input 1`] = `
<div
  className="movie-wrapper col-lg-7"
>
  <div
    className="movie"
  >
    <p
      className="movie-name"
    >
      Name: 
      Mock movie
    </p>
    <p
      className="movie-genre"
    >
      Genre: 
      unknown
    </p>
    <p
      className="movie-year"
    >
      Year: 
      2002
    </p>
  </div>
  <div
    className="movie"
  >
    <p
      className="movie-name"
    >
      Name: 
      Mock movie
    </p>
    <p
      className="movie-genre"
    >
      Genre: 
      unknown
    </p>
    <p
      className="movie-year"
    >
      Year: 
      2002
    </p>
  </div>
</div>
`;
